x-pack/plugins/security_solution/server/lib/detection_engine/rules/types.ts

```diff
@@ -158,6 +159,7 @@ export interface CreateRulesOptions {
   ruleId: RuleId;
   immutable: Immutable;
   index: IndexOrUndefined;
+  dataViewId?: DataViewIdOrUndefined;
```

Comment:
```suggestion
  dataViewId: DataViewIdOrUndefined;
```
Making this type non-optional gives better assurance that it isn't accidentally omitted from the list of parameters when `createRules` is called. This will help catch some of the cases in other comments where `dataViewId` is not wired through all the routes that use `createRules`.

Reply:
resolved here: [`2ce7c44` (#130929)](https://github.com/elastic/kibana/pull/130929/commits/2ce7c440573964a598730a124478cb7ed29d5d88)